src/wazuh_modules/vulnerability_detector/wm_vuln_detector_nvd.c

```diff
@@ -3187,10 +3187,15 @@ int wm_vuldet_add_dic_cpe(sqlite3 *db, int *cpe_index, cpe_list **node_list, cpe
 
     for (cpe_it = g_app, pos = 0; cpe_it; cpe_it = cpe_it->next, pos++) {
         cpe_it->pos = pos;
-        if (wm_vuldet_add_cpe(cpe_it, NULL, *node_list, *cpe_index)) {
-            mtdebug2(WM_VULNDETECTOR_LOGTAG, VU_AGENT_CPE_PARSE_ERROR, cpe_it->raw, o_app->id);
+        // The generation of CPEs where the version has not been extracted and is null is avoided.
+        if (strncmp(cpe_it->version, "0", 1)) {
```

Comment:
Here we need to use `strcmp(cpe_it->version, "0")`. Otherwise, only the first character is compared so versions like `0.8.3` will be evaluated as 0.

Reply:
Done in https://github.com/wazuh/wazuh/pull/14356/commits/dfc6aec358d99273a500edecddddca87360f2ccb